src/services/loopd/loop-service.ts

```diff
@@ -0,0 +1,255 @@
+    swapClient.loopOut.bind(swapClient),
+  )
+  const clientSwapOutQuote = util.promisify<QuoteRequest, OutQuoteResponse>(
+    swapClient.loopOutQuote.bind(swapClient),
+  )
+  const clientSwapOutTerms = util.promisify<TermsRequest, OutTermsResponse>(
+    swapClient.loopOutTerms.bind(swapClient),
+  )
+
+  const healthCheck = async (): Promise<boolean | SwapServiceError> => {
+    try {
+      const request = new TokensRequest()
+      const resp = await clientHealthCheck(request)
+      if (resp) return true
+    } catch (error) {
+      return new SwapErrorHealthCheckFailed(error)
+    }
+    return false
+  }
+
+  const swapOut = async function ({
+    amount,
+    maxSwapFee,
+    swapDestAddress,
+  }: SwapOutArgs): Promise<SwapOutResult | SwapServiceError> {
+    const fee = maxSwapFee ? maxSwapFee : 20000
+    try {
+      const request = new LoopOutRequest()
+      // on regtest, set the publication deadline to 0 for faster swaps, otherwise
+      // set it to 30 minutes in the future to reduce swap fees
+      const thirtyMins = 30 * 60 * 1000
+      const swapPublicationDeadline =
+        btcNetwork === BtcNetwork.regtest ? 0 : Date.now() + thirtyMins
+      if (swapDestAddress) request.setDest(swapDestAddress)
+      request.setAmt(amount)
+      request.setMaxSwapFee(fee)
+      request.setMaxPrepayRoutingFee(fee)
+      request.setMaxSwapFee(fee)
+      request.setMaxPrepayAmt(fee)
+      request.setMaxMinerFee(fee)
+      request.setSwapPublicationDeadline(swapPublicationDeadline)
+      request.setInitiator(`galoy-${loopdInstanceName}`)
+      if (btcNetwork === BtcNetwork.regtest) request.setHtlcConfirmations(1)
+      if (btcNetwork === BtcNetwork.regtest) request.setSweepConfTarget(2)
+      const resp = await clientSwapOut(request)
+      const swapOutResult: SwapOutResult = {
+        htlcAddress: resp.getHtlcAddress(),
+        serverMessage: resp.getServerMessage(),
+        swapId: resp.getId(),
+        swapIdBytes: resp.getIdBytes().toString(),
+      }
+      return swapOutResult
+    } catch (error) {
+      if (error.message.includes("channel balance too low")) {
+        return new SwapErrorChannelBalanceTooLow(error)
+      }
+      return new SwapServiceError(error)
+    }
+  }
+
+  const swapListener = function (): SwapClientReadableStream<SwapListenerResponse> {
+    try {
+      const request = new MonitorRequest()
+      const listener = swapClient.monitor(request)
+      // listener.on("data" | "status" | "end")
+      listener.on("data", (data: SwapStatus & SwapStatusResultWrapper) => {
+        listener.pause()
+        // parse data to our interface
+        const stateVal = data.getState()
+        let state
+        try {
+          state = Object.keys(SwapState).find((key) => {
+            // eslint-disable-next-line
+            // @ts-ignore
+            return SwapState[key] === stateVal
+          })
+        } catch (e) {
+          state = SwapState.FAILED
```

Comment:
is there open telemetry set up for those errors?

Reply:
Yeah returning SwapState.Failed will be picked up by the swap-listener and wrap it in a span see https://github.com/GaloyMoney/galoy/blob/37052f0cbe3cb6ca663dec8e2f7158d86edf2b1a/src/app/swap/swap-listener.ts#L47